lightning/src/ln/outbound_payment.rs

```diff
@@ -1608,7 +1624,9 @@ impl OutboundPayments {
 				entry.insert(PendingOutboundPayment::AwaitingInvoice {
 					expiration,
 					retry_strategy,
-					max_total_routing_fee_msat,
+					route_params_config,
+					// Retained for downgrade support.
+					max_total_routing_fee_msat: None,
```

Comment:
If you set this to `None`, then downgrades will not retain the value since they don't know how to read `route_params_config`. We need to set both in order to support downgrades.

Reply:
Ah, right! I was not writing that so that we can phase it out in future versions because it will be unused. But now I understand that we would still need to write it so that older versions can find and read this value.
Thanks a lot for the pointer!
Fixed in [pr3342.06](https://github.com/shaavan/rust-lightning/commits/pr3342.06)